### DISCUSSION TITLE
Send data to all requests to server

### DISCUSSION THREAD

**Discussion by sarik-k** [inertiajs/inertia#1650] [Help] [answered]
How do I send a piece of data in each and every request made to server?

Let's say I want to send {foo: "bar"} in each and every request, how do I do it? It tried the before event, but I can't figure out how to modify outgoing data. I'm using Laravel/Vue 3

JS: 
"@inertiajs/inertia": "^0.11.0",
"@inertiajs/inertia-vue3": "^0.6.0",

PHP: 
"inertiajs/inertia-laravel": "^0.6.3",

**Comment by reinink** [comment, marked as answer]
Hey! You can do this like this:

```js
import { router } from '@inertiajs/vue3'

router.on('before', (event) => {
  event.detail.visit.data.foo = 'bar'
})
```

But I would discourage doing this, as for GET requests this is going to end up being `/endpoint?foo=bar`, and for `POST/PUT/PATCH` it will be part of the body.

I think it's better to send a header instead:

```js
import { router } from '@inertiajs/vue3'

router.on('before', (event) => {
  event.detail.visit.headers.foo = 'bar'
})
```

Hope that helps.

**Reply by sarik-k** [reply]
Thanks, I think the first method works as I need it as part of the request.
